src/test_application/OSPMagnum.cpp

```diff
@@ -159,71 +161,195 @@ osp::active::ActiveScene& OSPMagnum::scene_create(std::string && name)
+	{"G", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::G  }},
+	{"H", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::H  }},
+	{"I", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::I  }},
+	{"J", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::J  }},
+	{"K", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::K  }},
+	{"L", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::L  }},
+	{"M", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::M  }},
+	{"N", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::N  }},
+	{"O", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::O  }},
+	{"P", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::P  }},
+	{"Q", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::Q  }},
+	{"R", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::R  }},
+	{"S", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::S  }},
+	{"T", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::T  }},
+	{"U", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::U  }},
+	{"V", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::V  }},
+	{"W", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::W  }},
+	{"X", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::X  }},
+	{"Y", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::Y  }},
+	{"Z", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::Z  }},
+
+	//Number keys
+	{"0", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::NumZero  }},
+	{"1", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::NumOne  }},
+	{"2", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::NumTwo  }},
+	{"3", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::NumThree  }},
+	{"4", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::NumFour  }},
+	{"5", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::NumFive  }},
+	{"6", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::NumSix  }},
+	{"7", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::NumSeven  }},
+	{"8", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::NumEight  }},
+	{"9", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::NumNine  }},
+
+	//Function keys
+	{"F1", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::F1  }},
+	{"F2", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::F2  }},
+	{"F3", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::F3  }},
+	{"F4", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::F4  }},
+	{"F5", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::F5  }},
+	{"F6", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::F6  }},
+	{"F7", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::F7  }},
+	{"F8", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::F8  }},
+	{"F9", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::F9  }},
+	{"F10", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::F10  }},
+	{"F11", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::F11  }},
+	{"F12", {sc_keyboard, (int)OSPMagnum::KeyEvent::Key::F12  }},
+
+	//Mouse
+	{"RMouse", {sc_mouse, (int)OSPMagnum::MouseEvent::Button::Right }},
+	{"LMouse", {sc_mouse, (int)OSPMagnum::MouseEvent::Button::Left }},
+	{"MMouse", {sc_mouse, (int)OSPMagnum::MouseEvent::Button::Middle }}
+};
+
+std::vector<ButtonVarConfig> parse_control(std::string_view str) {
+	//Split string
+	std::string s(str);
```

Comment:
Why use a string view when it's just copied into a string? String views work best if all you're doing is substr, which seems to be the case here.